### ISSUE TITLE
root execute.

### ISSUE THREAD

**Issue by gintering1000** [Homebrew/brew#12841]
### Provide a detailed description of the proposed feature

i use homebrew with root.
- delete [ check-run-command-as-root () ]
- change chown root:wheel /usr/local/Homebrew

i know homebrew have good security.
but can you add change option for using root
ex : brew setting --rootuse enable

cuz of this function ( [ check-run-command-as-root () ] ), it is hard to make automatic computer preference after install osx.
( everytime, i have to modify brew.sh file.. )

### What is the motivation for the feature?

basically so many linux package manager can install server. 
so.. technical user need root finally.
when install osx first, without sed, awk~ user have to modify brew.sh file with vim.

### How will the feature be relevant to at least 90% of Homebrew users?

almost server daemon package is for root.

### What alternatives to the feature have been considered?

user have to modify brew.sh file directly ( not automatically with command )

i wish user can control what they want.

**Comment by MikeMcQuaid** [comment]
No, sorry, we're not interested in this.
